Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -90,6 +90,22 @@ extension File {
         }
     }
 
+    public func syntaxKindsByLine(startLine: Int? = nil,
+                                  endLine: Int? = nil) -> [(Int, [SyntaxKind])] {
+        let contents = self.contents as NSString
+        let kindsWithLines = syntaxMap.tokens.map { token -> (Int, SyntaxKind) in
+            let tokenLine = contents.lineAndCharacterForByteOffset(token.offset)
+            return (tokenLine!.line, SyntaxKind(rawValue: token.type)!)
+        }.filter { line, token in
+            return line >= (startLine ?? 0) && line <= (endLine ?? Int.max)
```

Comment:
> It needs extending SourceKitten for implementing that.

I meant that for best performance implementation.

For better performance implementation.
`startLine` and `endLine` calculation is not needed in `validateFile`. It would be better filtering by `bodyOffset` and `bodyLength` as like [here](https://github.com/realm/SwiftLint/blob/master/Source%2FSwiftLintFramework%2FExtensions%2FFile%2BSwiftLint.swift#L85-L88).

Calling `lineAndCharacterForByteOffset` is heavy.
With this PR, the duration of linting KeychainAccess on my Mac using Instruments is 6159ms.
`lineAndCharacterForByteOffset` uses 2867ms on that:
<img width="836" alt="screenshot 2016-01-12 00 44 06" src="https://cloud.githubusercontent.com/assets/33430/12238111/ae0533cc-b8c5-11e5-9d47-39cafd43da39.png">
I think we can reduce calls to `lineAndCharacterForByteOffset` by that. :wink: